### DISCUSSION TITLE
[RosettaAPI] How to know the block height of a transaction

### DISCUSSION THREAD

**Discussion by Ljzn** [hiero-ledger/hiero-mirror-node#2924] [Help] [answered]
Hi, I'm using the rosetta api to monitoring transactions. 

The "block" is a special concept of Rosetta, that not exist on Hedera explorer. How to get the block height of a transaction?

There would be such case: a user give us a txid, said he deposited some coin to our address. So we need to refetch the tx data from rosetta api, but it need the block identifier at "/block/transaction/" route.

I understand the raw tx can be got from rest api, but it would be easier to integrate if we can do it on rosetta api.

The solutions maybe like:
- implement the search api https://www.rosetta-api.org/docs/SearchApi.html
- or a way to know the block identifier of any tx

**Comment by xin-hedera** [comment, marked as answer]
can you create a ticket? we can implement the `/search/transactions` endpoint to support search by rosetta transaction identifier (hash) as the first step, and add support of other search criteria later.
